src/main/java/org/hisp/dhis/antlr/function/AntlrFunctionLog.java

```diff
@@ -0,0 +1,57 @@
+package org.hisp.dhis.antlr.function;
+
+/*
+ * Copyright (c) 2004-2019, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import org.hisp.dhis.antlr.operator.AntlrComputeFunction;
+
+import java.util.List;
+
+import static org.hisp.dhis.antlr.AntlrParserUtils.castDouble;
+
+/**
+ * Math function: log [natural logarithm, base e]
+ *
+ * @author Jim Grace
+ */
+public class AntlrFunctionLog
+    extends AntlrComputeFunction
+{
+    @Override
+    public Object compute( List<Object> values )
+    {
+        if ( values.size() == 1 )
+        {
+            return Math.log( castDouble( values.get( 0 ) ) );
+        }
+        else
```

Comment:
if `values` list is not null but empty then else block will be executed. can we make it else if ( values.size() > 1 ) to make sure we execute else block if list has more than 1 item?

Reply:
We could do that, but the parser guarantees that we must have at least 1 argument, which will result in at least 1 value.